spinalcordtoolbox/reports/slice.py

```diff
@@ -444,3 +456,44 @@ def get_center(self, img_idx=-1):
         size_y = self.axial_dim(image)
         size_x = self.coronal_dim(image)
         return np.ones(dim) * size_x / 2, np.ones(dim) * size_y / 2
+
+    def mosaic(self, nb_column=0, dilation=(0, 0, 0)):
+        """Obtain matrices of the mosaics
+
+        Mosaic images are cropped based on the bounding box of the spinal cord segmentation.
+
+        :param nb_column: number of mosaic columns
+        :param dilation: a tuple containing (x,y,z) integers for dilating the bounding box used for cropping.
+        :return: tuple of numpy.ndarray containing the mosaics of each slice pixels
+        """
+        # 0. Use the segmentation image to initialize the image cropper
+        self._image_seg = self._images[-1]
+        cropper = ImageCropper()
+        cropper.get_bbox_from_mask(self._image_seg)
+
+        # 1. Compute the sizes of the patches, as well as the overall image
+        # 1a. Compute width and height of mosaic squares. (This is assumed to be a square for Axial slices.)
+        size_h = cropper.bbox.xmax - cropper.bbox.xmin + 1 + (2 * dilation[0])  # SAL -> SI axis provides height
+        size_w = cropper.bbox.ymax - cropper.bbox.ymin + 1 + (2 * dilation[1])  # SAL -> AP axis provides width
+        # 1b. Calculate number of columns to display on the report. (By default, size=15 -> 30x30 squares -> 20 columns)
+        if nb_column == 0:
+            nb_column = 600 // size_w
+        # 1c. Calculate number of rows to display.
+        nb_slices = cropper.bbox.zmax - cropper.bbox.zmin + 1 + (2 * dilation[2])  # SAL -> LR axis provides nb_slices
+        nb_row = math.ceil(nb_slices // nb_column) + 1
+        # 1d. Compute the matrix size of the overall mosaic image
+        matrix_sz = (int(size_h * nb_row), int(size_w * nb_column))
+
+        # 2. Use the previously-defined segmentation-based cropper to crop the images to the necessary size
+        matrices = list()
+        for image in self._images:
+            image_cropped = cropper.crop(img_in=image, dilate=dilation)
```

Comment:
As mentioned in commit 8ce10aa8f980a2ffc5c9d3f1e94affff2763e00a, using the `ImageCropper` class doesn't really mesh well with the `Axial` approach of implementing a `self.crop` method.

So, I'm opening this thread just to discuss how we should design this approach: Should we make an effort to keep the cropping methods consistent between all `Slice` subclasses?

Reply:
Is `self.crop` even really part of the public API for `Slice`? Also, it's a `@staticmethod`, so it could very well be a standalone function rather than a method on `self`. Anyway, I don't see much benefit to trying for consistency here.